### PR TITLE
Add gongo bot base functionality

### PR DESCRIPTION
**Pull request by danielsp45**
In this commit the basic belt attributions were created using the discord.command instead of the docopt library.
The bot checks if the Belt can be attributed to the Ninja (i.e. a Black belt can't be attributed to a ninja with a Orange belt), if it verifies that the belt can be attributed, the bot adds the new belt and removes the old one.

### CODE REVIEW

File: bot/client.py [hunk 1]
```diff
@@ -1,42 +1,82 @@
...
+import traceback
 
 from bot import VERSION
 
+client = commands.Bot(command_prefix=commands.when_mentioned_or('$'),
+                      help_command=None,
+                      intents=discord.Intents.all(),
+                      case_insensitive=True)
+
+cinturoes = [
```
Comment (nelsonmestevao) [inline bot/client.py:14]: What do you think about creating an Enum class for this?

**Review by nelsonmestevao — COMMENTED** [review]
Seems really good work. :clap: 

Can I ask review from @RuiL1904 as well?

File: bot/client.py (hunk 1 above)
Comment (danielsp45) [reply bot/client.py:14]: I think it is a good idea, Rui also suggested it to me and I forgot to implement it. I'll change it as soon as I can!

**Comment by danielsp45** [comment]
> 

Thank you, and yes of course.

I also wanted to ask about one thing. The way I did the belt attribution doesn't allow to give a belt to someone without one, therefore I wanted to ask if every new ninja that arrives to the server is going to be given a white belt or it has to be manually given by a mentor?

File: bot/client.py [hunk 2]
```diff
@@ -1,42 +1,82 @@
-"""Gongo.
-
-Usage:
-  Gongo (promove | p) <user> a <belt>
-
-"""
+from tabnanny import check
+from unicodedata import name
 import discord
-from docopt import docopt, DocoptExit
```
Comment (nelsonmestevao) [inline bot/client.py:8]: Since we don't need it any more, I think we should remove it from `requirements.txt` as well its peer dependencies.

**Comment by nelsonmestevao** [comment]
> The way I did the belt attribution doesn't allow to give a belt to someone without one, therefore I wanted to ask if every new ninja that arrives to the server is going to be given a white belt or it has to be manually given by a mentor?

Oh, I see. Yes, that might be a problem for two reasons.

1. Someone could forget to give the base badge to a ninja which would not be a problem if the bot just says clearly the problem. 
2. The real problem, it is a valid state to a ninja not having a belt. Only after its second session a ninja receives its first belt (the white one).

File: bot/client.py [hunk 3]
```diff
@@ -1,42 +1,82 @@
...
 
 from bot import VERSION
 
+client = commands.Bot(command_prefix=commands.when_mentioned_or('$'),
+                      help_command=None,
+                      intents=discord.Intents.all(),
+                      case_insensitive=True)
+
+cinturoes = [
+    "Branco", "Amarelo", "Laranja", "Verde", "Azul", "Vermelho", "Roxo",
```
Comment (nelsonmestevao) [inline bot/client.py:15]: The order is not right.

I leave here some help from our flyers. 

- [belts-2020.pdf](https://github.com/coderdojobraga/gongo/files/8502024/belts-2020.pdf)

- [belts-2017.pdf](https://github.com/coderdojobraga/gongo/files/8502022/belts-2017.pdf)

**Review by RuiL1904 — APPROVED** [review]
Great work! 👏🏼
The class `Ninja_data` can be refactored in some ways, but for now let's leave it like this.

**Comment by danielsp45** [comment]
> > The way I did the belt attribution doesn't allow to give a belt to someone without one, therefore I wanted to ask if every new ninja that arrives to the server is going to be given a white belt or it has to be manually given by a mentor?
> 
> Oh, I see. Yes, that might be a problem for two reasons.
> 
> 1. Someone could forget to give the base badge to a ninja which would not be a problem if the bot just says clearly the problem.
> 2. The real problem, it is a valid state to a ninja not having a belt. Only after its second session a ninja receives its first belt (the white one).

So should I just make it possible to attribute the white belt for now and think of a possible solution for those problems after?

**Comment by nelsonmestevao** [comment]
> So should I just make it possible to attribute the white belt for now and think of a possible solution for those problems after?

@danielsp45 I think being able to give the white belt to a ninja should be sufficient :+1:

**Comment by danielsp45** [comment]
I added a new pull request by mistake. All the changes of that pull request are now here

**Review by nelsonmestevao — APPROVED** [review]
Feel free to squash and merge 🤘

**Pull request merged**